Testscripts/Windows/VALIDATE-DA.ps1

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+<#
+.Synopsis
+    This script validates Dependency Agent.
+
+.Description
+    The script will install the Latest Dependency Agent and validate install/uninstall
+    scenarios as well as enable/disable scenarios.
+'
+#>
+param([String] $TestParams,
+      [object] $AllVmData)
+function Main {
+    param (
+        $VMName,
+        $Ipv4,
+        $VMPort,
+        $VMUserName,
+        $VMPassword
+    )
+
+    $remoteScript = "validate-da.sh"
+    #######################################################################
+    #
+    #	Main body script
+    #
+    #######################################################################
+
+    # Checking the input arguments
+    if (-not $VMName) {
+        Write-LogErr "VM name is null!"
+        return "FAIL"
+    }
+
+    Run-LinuxCmd -username $VMUserName -password $VMPassword -ip $Ipv4 -port $VMPort $remoteScript -runAsSudo
+    $testResult = Collect-TestLogs -LogsDestination $LogDir -ScriptName $remoteScript.split(".")[0] -TestType "sh" `
```

Comment:
Generally, code that starts from parts and assembles them into a composite object is easier to write, maintain, and is less likely to be buggy.

Recommendation: Rather than starting with `$remoteScript` and decomposing it to get the two parts, have two variables `$remoteScriptName` and `$remoteScriptType`. Concatenate them where needed for the call to `Run-LinuxCmd`.

Reply:
Done